src/lib/core/a11y/focus-trap.ts

```diff
@@ -114,4 +158,81 @@ export class FocusTrap {
 
     return null;
   }
+
+  /** Creates an anchor element. */
+  private _createAnchor(): HTMLElement {
+    let anchor = document.createElement('div');
+    anchor.tabIndex = this._enabled ? 0 : -1;
+    anchor.classList.add('cdk-visually-hidden');
+    anchor.classList.add('cdk-focus-trap-anchor');
+    return anchor;
+  }
+}
+
+
+/** Factory that allows easy instantiation of focus traps. */
+@Injectable()
+export class FocusTrapFactory {
+  constructor(private _checker: InteractivityChecker, private _ngZone: NgZone) { }
+
+  create(element: HTMLElement, deferAnchors = false): FocusTrap {
+    return new FocusTrap(element, this._checker, this._ngZone, deferAnchors);
+  }
+}
+
+
+/**
+ * Directive for trapping focus within a region.
+ * @deprecated
+ */
+@Directive({
+  selector: 'cdk-focus-trap',
```

Comment:
focus-trap has been deprecated, so i think its fine to remove, but now cdk-focus-trap is deprectaed too, since we're moving to an attribute version